### ISSUE TITLE
Releases failing due to build cleanup and rate limiting

### ISSUE THREAD

**Issue by nzoschke** [convox/rack#855]
The recently introduced old build cleanup is causing side effects on existing apps. If the apps have lots (say 100s) of builds, the build cleanup itself errors out due to rate limiting, then in-progress builds and releases can also error due to rate limiting.

The build cleanup needs to be revised to be rate limit aware.

Right now, affected apps can get unstuck by manually cleanup up old builds with a shell script:

```bash
convox api get /apps/myapp/builds?limit=500 | tee builds.json && jq -r '. | sort_by(.started) | .[0:-50] | .[].id' builds.json | xargs -L1 -P1 convox builds delete --app myapp
```

**Comment by nzoschke** [comment]
A fix for this shouldn't be too hard. Instead of deleting all the old builds in parallel it can delete them one at a time since its a background task.

**Comment by MiguelMoll** [comment]
Fix went out with #918
